zerver/views/auth.py

```diff
@@ -337,10 +388,18 @@ def log_into_subdomain(request):
 
     email_address = data['email']
     full_name = data['name']
-    user_profile, return_data = authenticate_remote_user(request, email_address)
-    invalid_subdomain = bool(return_data.get('invalid_subdomain'))
-    return login_or_register_remote_user(request, email_address, user_profile,
-                                         full_name, invalid_subdomain)
+    is_signup = data['is_signup']
+    if is_signup:
+        return maybe_send_to_registration(request, email_address, full_name)
```

Comment:
What happens if the user is trying to register with an email that already has an account, here?